### PR TITLE
Add tf12/tg19 upgrade guide

### PR DESCRIPTION
**Pull request by yorinasub17**
This is a port of https://github.com/gruntwork-io/infrastructure-live-multi-account-acme/pull/22 into the docs site.

### CODE REVIEW

File: content/guides/upgrading-to-tf12-tg19/index.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+This guide walks you through the steps you should take when upgrading your Reference Architecture deployment to
+Terraform 0.12.x and Terragrunt 0.19.x.
+
+## Background
+
+Terraform 0.12 (referred to as TF12) was released in May, 2019, and it included a few major changes:
+
+1. More strict rules around what can go in a `.tfvars` file. In particular, any variable defined in a `.tfvars` file
+   that does not match a corresponding `variable` definition in your `.tf` files produces an error.
+1. A shift from HCL to HCL2 as the main syntax. This included support for first-class expressions (i.e., using variables
```
Comment (brikis98) [inline content/guides/upgrading-to-tf12-tg19/index.md]: Should probably go first :)

File: content/guides/upgrading-to-tf12-tg19/index.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+[here](https://github.com/gruntwork-io/terragrunt/releases).
+
+
+#### Migrate terraform.tfvars to terragrunt.hcl
+
+Once you have terragrunt 0.19 available on your machine, you will need to migrate your `terraform.tfvars` files to
+`terragrunt.hcl` files before you can start to use the new version. You will need to follow the steps outlined in the
+[Terragrunt 0.19.x migration
+guide](https://github.com/gruntwork-io/terragrunt/blob/master/_docs/migration_guides/upgrading_to_terragrunt_0.19.x.md)
+for each of your live config (e.g in your `infrastructure-live` repo).
```
Comment (brikis98) [inline content/guides/upgrading-to-tf12-tg19/index.md]: GitHub's diff view shows all of the text from here onward in italics. GitHub bug or extra `_` somewhere?

File: content/guides/upgrading-to-tf12-tg19/index.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+You should do this in parallel with each module upgrade. For example, if you were upgrading the module
+`infrastructure-modules/networking/vpc-app`, you should upgrade the live config that deploys that module to a pre-prod
+environment to test the changes you are making to that module (e.g
+[`infrastructure-live/dev/us-east-1/dev/vpc`](https://github.com/gruntwork-io/infrastructure-live-multi-account-acme/tree/dd6dce7f737f8c1bd32466b69e905b2bdd25db80/dev/us-east-1/dev/vpc)).
+The rough process should be:
+
+1. Upgrade the module to TF12 syntax following the steps [listed above](#upgrade-each-module-for-0-12-compatibility).
+1. Push the changes to a branch for testing.
```
Comment (brikis98) [inline content/guides/upgrading-to-tf12-tg19/index.md]: No commits/pushing for local or pre-prod testing. Instead, use `--terragrunt-source` or `TERRAGRUNT_SOURCE` env var to test with local checkout. Only commit/push/release _after_ local testing, when you are ready to roll the changes to prod envs.

File: content/guides/upgrading-to-tf12-tg19/index.md (hunk 2 above)
Comment (yorinasub17) [reply content/guides/upgrading-to-tf12-tg19/index.md]: Looking at the rendering, it's rendering italics based on the underscores in the URL link above :(

File: content/guides/upgrading-to-tf12-tg19/index.md (hunk 1 above)
Comment (yorinasub17) [reply content/guides/upgrading-to-tf12-tg19/index.md]: Good point. Done!

File: content/guides/upgrading-to-tf12-tg19/index.md (hunk 3 above)
Comment (yorinasub17) [reply content/guides/upgrading-to-tf12-tg19/index.md]: Good point. Done!

**Comment by yorinasub17** [comment]
Ok incorporated all the feedback. Going to merge and release this now. Thanks for the review!

**Pull request merged**